app/src/main/java/io/neurolab/main/TestModeActivity.java

```diff
@@ -100,6 +109,28 @@ public void onReceive(Context context, Intent intent) {
         };
     };
 
+    private void feedConfigSetToArduino(){
+        // Grabbing the application shared preference and the values. Have added the default values for the edge case when user
+        // lets say starts up the application for the first time and directly goes to the Test Mode Activity. Shared preferences would
+        // not be initialized in that case.
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
+
+        String channels = sharedPreferences.getString(NeuroSettingsFragment.KEY_CHANNELS, Integer.toString(NUM_CHANNELS_DEFAULT));
+        String samplesPerSecond = sharedPreferences.getString(NeuroSettingsFragment.KEY_SAMPLES, Integer.toString(SAMPLES_PER_SECOND_DEFAULT));
+        String bins = sharedPreferences.getString(NeuroSettingsFragment.KEY_BINS, Integer.toString(BIN_NUMBER_DEFAULT));
+
+        // Adding space as a delimeter for arduino to parse them individually.
+        String configString = channels + " " + samplesPerSecond + " " + bins;
+
+        // Writing the config string to arduino
+        serialPort.write(configString.getBytes());
+
+        // Toast for validation
+        Toast.makeText(this, "The following configuration is fed to Arduino: " +
```

Comment:
We can set this message in a snackbar. What do you think?

Reply:
@CloudyPadmal yeah. It's a good suggestion. Better suited for such info. Also can be swiped, unlike Toast. Thanks. Will update in a min.